contracts/Recruitment.sol

```diff
@@ -72,7 +84,143 @@ contract Recruitment {
+  */
+  function getCandidate(address wallet) external view returns(FrontDoorStructs.Candidate memory) {
+    return candidateList[wallet];
+  }
+
+  /**
+    * @notice Registers a referrer with their email
+    * @param email The email of the referrer to be registered.
+  */
+  function registerReferrer(string memory email) external {
+    FrontDoorStructs.Referrer memory referrer = FrontDoorStructs.Referrer(msg.sender, email,0);
+    referrerList[msg.sender] = referrer;
+  }
+
+  /**
+    * @notice Getter function of Referrer.
+    * @param wallet The wallet address of a referrer to be fetched from the mapping
+    * @return Referrer The type of FrontDoorStruct.Referrer 
+  */
+  function getReferrer(address wallet) external view returns(FrontDoorStructs.Referrer memory) {
+    return referrerList[wallet];
+  }
+
+  /**
+    * @notice Registers a referee with their email
+    * @param email The email of the referee to be registered.
+  */
+  function registerReferee(string memory email) external {
+    FrontDoorStructs.Referee memory referee = FrontDoorStructs.Referee(msg.sender, email,0);
+    refereeList[msg.sender] = referee;
+  }
+
+  /**
+    * @notice Getter function of Referee.
+    * @param wallet The wallet address of a referee to be fetched from the mapping
+    * @return Referee The type of FrontDoorStruct.Referee 
+  */
+  function getReferee(address wallet) external view returns(FrontDoorStructs.Referee memory) {
+    return refereeList[wallet];
+  }
+
+  /**
+    * @notice Registers a job
+    * @param bounty The amount of bounty set by the job poster.
+    * @param maxSalary The amount of maximum salary.
+    * @param minSalary The amount of minimum salary.
+    * @param companyName The name of company.
+    * @param location The location of the company.
+    * @param role The role for the job.
+    * @param jobURL The web URL of the job description.
+  */
+  function registerJob(uint256 bounty, uint256 maxSalary, uint256 minSalary, string memory companyName, string memory location, string memory role, string memory jobURL) external {
+    uint256 jobId = jobIdCounter.current();
+    FrontDoorStructs.Job memory job = FrontDoorStructs.Job(jobId, bounty, maxSalary, minSalary, companyName, location, role, jobURL);
+    jobList[jobId] = job;
+    jobIdCounter.increment();
+  }
+
+  /**
+    * @notice Getter function of Job.
+    * @param jobId The ID of a job to fetch from the mapping
+    * @return Job The type of FrontDoorStruct.Job 
+  */
+  function getJob(uint256 jobId) external view returns(FrontDoorStructs.Job memory){
+    return jobList[jobId];
+  }
+
+  /**
+    * @notice Registers a referral
+    * @param jobId The job ID already registered with the contract.
+    * @param refereeWallet The address of referee.
+    * @param refereeMail The email of the referee.
+  */
+  function registerReferral(uint256 jobId, address refereeWallet, string memory refereeMail) external {
```

Comment:
A bit of wrong requirement passed at first, but now that we have a candidate confirmation step where his/her address gets passed at that later stage then here we should get rid of candidate address. This because from a UX perspective it is a lot more likely the referrer does not know the candidates address. This way we should keep it input params as: (uint256 jobId, string memory refereeMail)

Reply:
okay, but in the data model, we are identifying a referee by their wallet address. If we remove this parameter here, then we will have to figure out an alternative way to add a referee (without their wallet address) with their email. We will have to add another mapping of referee with email.
I am on it now.

Reply:
So, I removed the `refereeWallet` and added `emailHash` instead.